dashboard/src/components/Config/AppRepoList/AppRepoControl.v2.tsx

```diff
@@ -26,11 +27,17 @@ export function AppRepoControl({
   const [refreshing, setRefreshing] = useState(false);
   const openModal = () => setModalOpen(true);
   const closeModal = () => setModalOpen(false);
-  const dispatch = useDispatch();
+  const dispatch: ThunkDispatch<IStoreState, null, Action> = useDispatch();
 
   const handleDeleteClick = (repoName: string, repoNamespace: string) => {
-    return () => {
-      dispatch(actions.repos.deleteRepo(repoName, repoNamespace));
+    return async () => {
+      await dispatch(actions.repos.deleteRepo(repoName, repoNamespace));
+      if (repoNamespace !== kubeappsNamespace) {
+        // Re-fetch repos in both namespaces
```

Comment:
Yes, I am a bit limited because I don't want to change the state definition or the calls done from the current UI. The thing is that calling `fetchRepos` cleans the current state. So if in the state, there are repos from `ns1` and `ns2` and we re-fetch only `ns2`, the repos from `ns1` will dissapear.

Added a comment to clarify it.